hoomd/md/pytest/test_constant_force.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2009-2022 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+import hoomd
+from hoomd.conftest import pickling_check, autotuned_kernel_parameter_check
+import numpy
+
+
+def test_attributes():
+    constant = hoomd.md.force.Constant(filter=hoomd.filter.All())
+
+    assert constant.constant_force['A'] == (0.0, 0.0, 0.0)
+    assert constant.constant_torque['A'] == (0.0, 0.0, 0.0)
+
+    constant.constant_force['A'] = (0.5, 0.0, 0.0)
+    assert constant.constant_force['A'] == (0.5, 0.0, 0.0)
+    constant.constant_force['A'] = (0.0, 0.0, 1.0)
+    assert constant.constant_force['A'] == (0.0, 0.0, 1.0)
+
+
+def test_attach(simulation_factory, two_particle_snapshot_factory):
+    constant = hoomd.md.force.Constant(filter=hoomd.filter.Type(['A']))
+
+    sim = simulation_factory(
+        two_particle_snapshot_factory(particle_types=['A', 'B'],
+                                      dimensions=3,
+                                      d=8))
+    integrator = hoomd.md.Integrator(.05)
+    integrator.methods.append(
+        hoomd.md.methods.Langevin(hoomd.filter.All(), kT=0))
+    integrator.forces.append(constant)
+    sim.operations.integrator = integrator
+    sim.run(0)
+
+    assert constant.constant_force['A'] == (0.0, 0.0, 0.0)
+    assert constant.constant_torque['A'] == (0.0, 0.0, 0.0)
+
+    constant.constant_force['A'] = (0.5, 0.0, 0.0)
+    assert constant.constant_force['A'] == (0.5, 0.0, 0.0)
+    constant.constant_torque['A'] = (0.0, 0.0, 1.0)
+    assert constant.constant_torque['A'] == (0.0, 0.0, 1.0)
+
+    sim.run(1)
+
+    if sim.device.communicator.rank == 0:
+        assert numpy.array_equal(constant.forces,
+                                 [[0.5, 0.0, 0.0], [0.5, 0.0, 0.0]])
+        assert numpy.array_equal(constant.torques,
+                                 [[0.0, 0.0, 1.0], [0.0, 0.0, 1.0]])
+
+
+def test_kernel_parameters(simulation_factory, two_particle_snapshot_factory):
+    constant = hoomd.md.force.Constant(filter=hoomd.filter.All())
+
+    sim = simulation_factory(two_particle_snapshot_factory(dimensions=3, d=8))
+    integrator = hoomd.md.Integrator(.05)
+    integrator.methods.append(
+        hoomd.md.methods.Langevin(hoomd.filter.All(), kT=0))
+    integrator.forces.append(constant)
+    sim.operations.integrator = integrator
+    sim.run(0)
+
+    def activate_kernel():
+        constant.constant_force['A'] = (1.0, 2.0, 3.0)
```

Comment:
I pushed this change to trigger the autotuner to run on every step of the test. Previously, the kernel only executed once.